app/main.py

```
@@ -1,3 +1,50 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        raise TypeError(f"Unsupported operand type for +: {type(other)}")
+
+    def __mul__(self, other: int | float) -> Distance:
+        return Distance(self.km * other)
+
+    def __iadd__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        if isinstance(other, (int, float)):
+            self.km += other
+        return self
```

This method doesn't handle unsupported operand types as required. If an unsupported type (like a string) is passed, it should raise a `TypeError`. Currently, it fails silently without modifying the object or raising an error. You can look at your `__add__` method for a good example of how to implement this.